internal/extractors/aggregator.go

```diff
@@ -0,0 +1,113 @@
+package extractors
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/innosat-mats/rac-extract-payload/internal/common"
+	"github.com/innosat-mats/rac-extract-payload/internal/innosat"
+)
+
+// Aggregator sorts and accumulates standalone and multi-packets
+func Aggregator(target chan<- common.DataRecord, source <-chan common.DataRecord) {
+	defer close(target)
+	multiPackBuffer := bytes.NewBuffer([]byte{})
+	var multiPackStarted bool
+	var multiPackStart common.DataRecord
+
+	for sourcePacket := range source {
+		if sourcePacket.Error != nil {
+			target <- sourcePacket
+			continue
+		}
+		reader := bytes.NewReader(sourcePacket.Buffer)
+		switch sourcePacket.SourceHeader.PacketSequenceControl.GroupingFlags() {
+		case innosat.SPStandalone:
+			// Produce error for unfinished multipack lingering
+			if multiPackStarted {
+				target <- makeUnfinishedMultiPackError(multiPackBuffer, sourcePacket)
+				multiPackBuffer.Reset()
+				multiPackStarted = false
+			}
+
+			// Report standalone pack
+			target <- sourcePacket
+		case innosat.SPStart:
+			// Produce error for unfinished multipack lingering
+			if multiPackStarted {
+				target <- makeUnfinishedMultiPackError(multiPackBuffer, sourcePacket)
+			}
+
+			// Start new multipack
+			multiPackStarted = true
+			multiPackBuffer.Reset()
+			multiPackStart = sourcePacket
+			_, err := io.Copy(multiPackBuffer, reader)
+			if err != nil && err != io.EOF {
+				sourcePacket.Error = err
+				target <- sourcePacket
+			}
+		case innosat.SPCont:
+			// Report error missing start packet
+			if !multiPackStarted {
+				sourcePacket.Error = errors.New("got continuation packet without a start packet")
+				target <- sourcePacket
+
+				multiPackStart = sourcePacket
+				multiPackStarted = true
+			}
+
+			// Concat SPCont packet
+			_, err := multiPackBuffer.ReadFrom(reader)
+			if err != nil && err != io.EOF {
+				sourcePacketCopy := sourcePacket
+				sourcePacketCopy.Error = err
+				target <- sourcePacketCopy
+			}
+		case innosat.SPStop:
+			// Report error missing start pack
+			if !multiPackStarted {
+				// Report error
+				sourcePacketCopy := sourcePacket
+				sourcePacketCopy.Error = errors.New("got stop packed without a start packet")
```

Comment:
```suggestion
				sourcePacketCopy.Error = errors.New("got stop packet without a start packet")
```

Also, why don't these get their own error functions?`No interesting data to trace, I guess?